### PR TITLE
Upgrade Misk to use wire 3.2.0 and construct gRPC clients with updated APIs

### PR DESCRIPTION
**Pull request by wesleyk**
To accommodate for updates made in Wire from https://github.com/square/wire/issues/1437

### CODE REVIEW

File: misk/src/main/kotlin/misk/client/GrpcClientModule.kt [hunk 1]
```diff
@@ -64,7 +67,8 @@ class GrpcClientModule<T : Service>(
           .baseUrl(baseUrl)
           .build()
 
-      return grpcClient.create(kclass)
+      // There should be *exactly one constructor* that takes in a grpcClient
+      return grpcClientClass.constructors.first().call(grpcClient)
```
Comment (wesleyk) [inline misk/src/main/kotlin/misk/client/GrpcClientModule.kt:71]: could probably add some sanity checks here if folks would find it useful; mainly that there's:
- One constructor
- One parameter on the constructor
- That parameter implements Service

Comment (swankjesse) [reply misk/src/main/kotlin/misk/client/GrpcClientModule.kt:71]: This should be fine.

Comment (swankjesse) [reply misk/src/main/kotlin/misk/client/GrpcClientModule.kt:71]: Looking at this and how it used to work, I think it was a mistake to take this API out of Wire in the first place. Wire can figure out what the naming pattern is and do the right thing, so the application never sees the implementation, only the interface.

FYI @egorand @oldergod

Comment (Egorand) [reply misk/src/main/kotlin/misk/client/GrpcClientModule.kt:71]: Yeah, let's definitely revisit.

Comment (swankjesse) [reply misk/src/main/kotlin/misk/client/GrpcClientModule.kt:71]: https://github.com/square/wire/pull/1527

**Pull request merged**